R/party.R

```diff
@@ -0,0 +1,299 @@
+#' @param teststat a character specifying the type of the test statistic to be
+#' applied.
+#' @param testtype a character specifying how to compute the distribution of
+#' the test statistic.
+#' @param mincriterion the value of the test statistic (for \code{testtype ==
+#' "Teststatistic"}), or 1 - p-value (for other values of \code{testtype}) that
+#' must be exceeded in order to implement a split.
+#' @param minsplit the minimum sum of weights in a node in order to be
+#' considered for splitting.
+#' @param maxdepth maximum depth of the tree. The default \code{maxdepth = 0}
+#' means that no restrictions are applied to tree sizes.
+#' @param ... Other options to pass to [party::ctree()].
+#' @return An object of class `BinaryTree-class`.
+#' library(survival)
+#' data(lung)
+#' cond_inference_surv_ctree(
+#'   Surv(time, status) ~ age + sex,
+#'   data = lung,
+#'   maxdepth = 2,
+#'   minsplit = 5
+#' )
+#' @seealso cond_inference_surv_cforest
+#' @export
+cond_inference_surv_ctree <-
+  function(formula,
+           data,
+           minsplit = 20,
+           maxdepth = 0,
+           teststat = "quad",
+           testtype = "Bonferroni",
+           mincriterion = 0.95,
+           ...) {
+    opts <- rlang::list2(...)
+
+    # Edit the control options to put some argument values into the controls
+    # function. This is a challenge since the control objects are S4 objects
+    # of S4 objects. If the top-level tuning parameter disagrees with an
+    # existing value, we over-write it an issue a warning.
+    if (any(names(opts) == "controls")) {
+      opts$controls <- resub_party_arg(opts$controls, "splitctrl", "minsplit", minsplit)
+      opts$controls <- resub_party_arg(opts$controls, "tgctrl",    "maxdepth", maxdepth)
+      opts$controls <- resub_party_arg(opts$controls, "varctrl",   "teststat", teststat)
+      opts$controls <- resub_party_arg(opts$controls, "gtctrl",    "testtype", testtype)
+      opts$controls <- resub_party_arg(opts$controls, "gtctrl",    "mincriterion", mincriterion)
+
+    } else {
+      opts$controls <-
+        rlang::call2(
+          "ctree_control",
+          .ns = "party",
+          !!!list(
+            minsplit = minsplit,
+            maxdepth = maxdepth,
+            teststat = teststat,
+            testtype = testtype,
+            mincriterion = mincriterion
+          )
+        )
+    }
+
+    tree_call <-
+      rlang::call2(
+        "ctree",
+        .ns = "party",
+        formula = rlang::expr(formula),
+        data = rlang::expr(data),!!!opts
+      )
+    rlang::eval_tidy(tree_call)
+  }
+
+# ------------------------------------------------------------------------------
+# some wrappers for ctree predictions
+
+approx_surv_fit <- function(x, .time = 0) {
+  dat <- stack_survfit_cph(x, n = 1)
+  interpolate_km_values(dat, .time)
+}
+
+#' @export
+#' @keywords internal
+#' @rdname party_internal
+surv_prob_ctree <- function(object, new_data, .time) {
```

Comment:
apart from `surv_prob_ctree()` and `surv_prob_cforest()` we also have `cph_survival_prob()` and `coxnet_survival_prop()`. Feels like we should standardize, how about `survival_prob_<fit class>()`?

Reply:
I agree we should standardize. Let us merge this as is and standardize after

Reply:
Thanks for finding these inconsistencies